app/models/mixins/verify_credentials_mixin.rb

```diff
@@ -0,0 +1,77 @@
+module VerifyCredentialsMixin
+  extend ActiveSupport::Concern
+
+  included do
+    class << self
+      Vmdb::Deprecation.deprecate_methods self, :validate_credentials_task => :verify_credentials_task
+    end
+  end
+
+  module ClassMethods
+    def validate_credentials_task(args, user_id, zone)
+      task_opts = {
+        :action => "Validate EMS Provider Credentials",
+        :userid => user_id
+      }
+
+      queue_opts = {
+        :args        => [*args],
+        :class_name  => name,
+        :method_name => "raw_connect?",
+        :queue_name  => "generic",
+        :role        => "ems_operations",
+        :zone        => zone
+      }
+
+      task_id = MiqTask.generic_action_with_callback(task_opts, queue_opts)
+      task = MiqTask.wait_for_taskid(task_id, :timeout => 30)
+
+      if task.nil?
+        error_message = "Task Error"
+      elsif MiqTask.status_error?(task.status) || MiqTask.status_timeout?(task.status)
+        error_message = task.message
+      end
+
+      [error_message.blank?, error_message]
+    end
+
+    def verify_credentials_task(userid, zone, options)
+      task_opts = {
+        :action => "Verify EMS Provider Credentials",
+        :userid => userid
+      }
+
+      encrypt_verify_credential_params!(options)
+
+      queue_opts = {
+        :args        => [options],
+        :class_name  => name,
+        :method_name => "verify_credentials?",
+        :queue_name  => "generic",
+        :role        => "ems_operations",
+        :zone        => zone
+      }
+
+      MiqTask.generic_action_with_callback(task_opts, queue_opts)
+    end
+
+    def verify_credentials?(args)
+      # Prevent the connection details, including the password, from being leaked into the logs
+      # and MiqQueue by only returning true/false
+      !!verify_credentials(args)
+    end
+
+    private
+
+    # Ensure that any passwords are encrypted before putting them onto the queue for any
+    # DDF fields which are a password type
+    def encrypt_verify_credential_params!(options)
+      params_for_create[:fields].each do |field|
```

Comment:
This is failing with an undefined method if called from `ExtManagementSystem`. My guess is that it's only applicable when an already existing provider is being verified, which is not very useful in our case. @Fryguy @agrare

Reply:
@skateman this is expected to be called on the provider class not on the base class.

Reply:
Thanks, this can be sorted out.